kmod/src/ioctl.h

```diff
@@ -454,11 +457,14 @@ struct scoutfs_ioctl_alloc_detail_entry {
  * EISDIR: either the source or destination is a directory.
  * ENODATA: either the source or destination file have offline extents.
  */
+#define SCOUTFS_IOC_MB_STAGE		(1 << 0)
+
 struct scoutfs_ioctl_move_blocks {
 	__u64 from_fd;
 	__u64 from_off;
 	__u64 len;
 	__u64 to_off;
+	__u64 flags;
```

Comment:
It occurs to me that if we're adding staging we need to also check the data_version so that we can't use move_blocks to insert stale old extents into the file if a write sneaks in before us.  We'd add it to the ioctl here, and after grabbing the i_mutex in move_blocks we'd make sure that the data_version ioctl arg matches the inode data_version if we're staging, and return -ESTALE if not.  Like the release ioctl does.